### ISSUE TITLE
Add support for the Azure IotHub fallback route

### ISSUE THREAD

**Issue by IvanovOleg** [terraform-providers/terraform-provider-azurerm#2719]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

When you create an IoT Hub with a custom route via terraform, the [IoTHub fallback route](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-messages-d2c#fallback-route) is disabled by default which is not suitable. If you enable this parameter in portal and run terraform again, it disables it.

### New or Affected Resource(s)

* azurerm_iothub

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

* #0000

**Comment by danielscholl** [comment]
The default action of creating a hub via cli or portal is to have the IoT fallback route enabled.  The default action of the terraform creating a hub is to have the IoT fallback route disabled with no option to enable it.

## Terraform Configuration

```
provider "azurerm" {
  version = "=1.21.0"
}

resource "azurerm_resource_group" "test" {
  name     = "IoTEdgeResources"
  location = "eastus2"
}

resource "azurerm_iothub" "test" {
  name                = "tf-hub"
  resource_group_name = "${azurerm_resource_group.test.name}"
  location            = "${azurerm_resource_group.test.location}"

  sku {
    name     = "S1"
    tier     = "Standard"
    capacity = "1"
  }
}
```

## Azure CLI Commands

```bash
# Create a Hub
az iot hub create --resource-group IoTEdgeResources --name test-cli-hub --sku S1 --location eastus2

# View fallbackRoute status
az iot hub list -g IoTEdgeResources --query \
  "[].{name:name, fallback:properties.routing.fallbackRoute.isEnabled}" \
  -otable
```

## Creation Result

```
Name             Fallback
-----------      ----------
test-cli-hub     True
test-tf-hub      False
```

**Comment by BlueBasher** [comment]
Terraform is indeed disabling the fallback route by default.
The difference is in the RoutingProperties, there should be a fallbackRoute and depending on a setting it should set isEnabled to true or false.

### Terraform created
```json
"routing": {
	"endpoints": {
		"serviceBusQueues": [],
		"serviceBusTopics": [],
		"eventHubs": [],
		"storageContainers": []
	},
	"routes": [],
	"fallbackRoute": {
		"name": "$fallback",
		"source": "DeviceMessages",
		"condition": "true",
		"endpointNames": [
			"events"
		],
		"isEnabled": false
	}
},
```

### Azure Portal created
```json
"routing": {
	"endpoints": {
		"serviceBusQueues": [],
		"serviceBusTopics": [],
		"eventHubs": [],
		"storageContainers": []
	},
	"routes": [],
	"fallbackRoute": {
		"name": "$fallback",
		"source": "DeviceMessages",
		"condition": "true",
		"endpointNames": [
			"events"
		],
		"isEnabled": true
	}
},
```

**Comment by StefanSchoof** [comment]
I think this issue can be closed, since #2764 is merged and is released with 1.23.0
